### PR TITLE
[Feature] 9c Player Download & Update

### PR DESCRIPTION
**Pull request by Atralupus**
Related to https://github.com/planetarium/9c-launcher/issues/1705
Related to https://github.com/planetarium/9c-launcher/issues/1707
Related to https://github.com/planetarium/9c-launcher/issues/1706

This PR will change the 9c player to install in the path '[userData](https://www.electronjs.org/docs/latest/api/app#appgetpathname)/player'

 - [x] Add `WindowsPlayerBinaryUrl`, `macOSPlayerBinaryUrl` in apv extra data
 - [x] Split launcher update, player update
 - [x] Implement 9c player download
 - [x] Replace update script
 - [x] Replace player execute code
 - [x] Fix lint...

### CODE REVIEW

**Comment by Atralupus** [comment]
Important thing is the scripts under the `update/*`

File: src/config.ts [hunk 1]
```diff
@@ -178,7 +178,10 @@ export function getBlockChainStorePath(): string {
 export const REQUIRED_DISK_SPACE = 20n * 1000n * 1000n * 1000n;
 export const SNAPSHOT_SAVE_PATH = app.getPath("userData");
 export const MAC_GAME_PATH = "9c.app/Contents/MacOS/9c";
```
Comment (BasixKOR) [inline src/config.ts:180]: This path should be edited too.

File: src/main/update/extract.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import extractZip from "extract-zip";
```
Comment (BasixKOR) [inline src/main/update/extract.ts:1]: Maybe we should consider some generic library for this: https://www.npmjs.com/package/decompress. I guess we can discuss this in another isssue.

File: src/main/update/extract.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    dir: tempDir,
+    onEntry: (_, zipfile) => {
+      const progress = zipfile.entriesRead / zipfile.entryCount;
+      win?.webContents.send("update extract progress", progress);
+    },
+  });
+  win.webContents.send("update extract complete");
+  console.log("[player] The zip archive", from, "has extracted to", tempDir);
+  win.webContents.send("update copying progress");
+  await utils.copyDir(tempDir, to);
```
Comment (BasixKOR) [inline src/main/update/extract.ts:27]: Is there a reason not to extract the target directory directly?

File: src/main/update/player-update.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  listeners: IUpdateOptions
+) {
+  const win = listeners.getWindow();
+
+  if (win === null) {
+    console.log("Stop update process because win is null.");
+    return;
+  }
+
+  win?.webContents.send("update download started");
```
Comment (BasixKOR) [inline src/main/update/player-update.ts:23]: I don't feel okay with reusing the update events for player updates again. As far as my knowledge of the update events, the current `updateMachine.ts` assumes all the steps are irreversible (i.e. you can't go back from extract to download)

File: src/config.ts (hunk 1 above)
Comment (Atralupus) [reply src/config.ts:180]: umm... my mistake😞

File: src/main/update/extract.ts (hunk 2 above)
Comment (Atralupus) [reply src/main/update/extract.ts:1]: In the first step, I did try to use the existing scripts
As you said, we should need to create a separate issue

File: src/main/update/extract.ts (hunk 3 above)
Comment (Atralupus) [reply src/main/update/extract.ts:27]: Maybe... Nope.
But as I mentioned in this [comment](https://github.com/planetarium/9c-launcher/pull/1727#discussion_r940002597), I tried to use the existing scripts, So I'll try to solve it another PR

File: src/main/update/player-update.ts (hunk 4 above)
Comment (Atralupus) [reply src/main/update/player-update.ts:23]: Thanks you, then... I'll solve to combine window send event

**Review by BasixKOR — APPROVED** [review]
Nitpicks included but seems fine.

**Pull request merged**